### DISCUSSION TITLE
How to do multiple prompts

### DISCUSSION THREAD

**Discussion by KodyAG** [v8hid/infinite-zoom-automatic1111-webui#20] [General]
How do i go about doing more than two prompts, for example 4? like if I wanted them to go in order and determining the length of each one?

**Comment by v8hid** [comment]
## First of all I want to explain some basics:
### To start, let's break down the workflow of the extension into three main steps:

- **Step 1: Choose an image to start with**
The program either generates an initial image using the first prompt you provide or you can upload your own image in the `custom initial image` field. This initial image will be the basis for the outpainting process.

- **Step 2: Generate outpaint steps**
Once you have your initial image, the program will start generating outpaint steps. The number of outpaint steps is determined by the `Total Outpaint Steps` input. In each outpaint step, the program makes the initial image smaller in the center of the canvas and generates a new image in the empty space that is created. This process is repeated for each outpaint step until the desired number is reached.

- **Step 3: Create a gradual zoom effect**
After all outpaint steps have been generated, the program creates an interpolation between each outpaint step to create a gradual zoom effect. The number of frames created between each outpaint step is determined by the `Zoom Speed` parameter and the `Frames per second` parameter.
`Number of frames` = `Zoom Speed` $\times$ `Frames per second`
also,
`length of each outpaint step` = `Number of frames` $\div$ `Frames per second` 

### About outpainting steps, you can use prompts to guide the generation process. For example, you can enter

| **outpaint step**  | **prompt** |
| ------------- | ------------- |
| 0 | Jungle with lush green trees and clear blue sky |
|  3 | Jungle with tall brown trees and orange sunset sky |
|  5 | Jungle with sparse trees and dark stormy sky  |
|  10 | Jungle with yellow autumn trees and purple twilight sky |

assuming `Total Outpaint Steps = 14`

Outpaint step of 0 to 3 uses "Jungle with lush green trees and clear blue sky"
Outpaint step of 3 to 5 uses "Jungle with tall brown trees and orange sunset sky"
Outpaint step of 5 to 10 uses "Jungle with sparse trees and dark stormy sky "
Outpaint step of 10 to 14 uses "Jungle with yellow autumn trees and purple twilight sky"

I hope this answered your question.
